easypermissions/src/main/java/pub/devrel/easypermissions/RationaleAppCompatDialogFragment.java

```diff
@@ -0,0 +1,114 @@
+package pub.devrel.easypermissions;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+
+import java.util.Arrays;
+
+import static pub.devrel.easypermissions.EasyPermissions.executePermissionsRequest;
+
+
+public class RationaleAppCompatDialogFragment extends DialogFragment implements Dialog.OnClickListener {
+
+    private EasyPermissions.PermissionCallbacks permissionCallbacks;
+
+    private int positiveButton;
+    private int negativeButton;
+    private int requestCode;
+    private String rationaleMsg;
+    private String[] permissions;
+
+    public static RationaleAppCompatDialogFragment newInstance(@StringRes int positiveButton,
```

Comment:
If these static methods are going to be public they should have JavaDoc (since they're part of the library API), if not then just make them package-protected or some non-public setting.